x-pack/plugins/ml/public/application/explorer/alerts_panel.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FC, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiBasicTableColumn,
+  EuiInMemoryTable,
+  EuiNotificationBadge,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FIELD_FORMAT_IDS } from '@kbn/field-formats-plugin/common';
+import useObservable from 'react-use/lib/useObservable';
+import { Axis, Chart, LineSeries, Position, ScaleType, Settings } from '@elastic/charts';
+import { CollapsiblePanel } from '../components/collapsible_panel';
+import { AnomalyDetectionAlert } from './anomaly_detection_alerts_state_service';
+import { useFieldFormatter } from '../contexts/kibana';
+import { useAnomalyExplorerContext } from './anomaly_explorer_context';
+import { Y_AXIS_LABEL_WIDTH } from './swimlane_annotation_container';
+
+export const AlertsPanel: FC = () => {
+  const [isOpen, setIsOpen] = useState(true);
+  const dateFormatter = useFieldFormatter(FIELD_FORMAT_IDS.DATE);
+  const durationFormatter = useFieldFormatter(FIELD_FORMAT_IDS.DURATION);
+
+  const { anomalyTimelineStateService, anomalyDetectionAlertsStateService } =
+    useAnomalyExplorerContext();
+
+  const alertsData = useObservable(anomalyDetectionAlertsStateService.anomalyDetectionAlerts$);
+  const selectedAlertsData = useObservable(anomalyDetectionAlertsStateService.selectedAlerts$);
+  const countByStatus = useObservable(anomalyDetectionAlertsStateService.countByStatus$);
+
+  const annotationXDomain = useObservable(anomalyTimelineStateService.timeDomain$);
+
+  const swimlaneContainerWidth = useObservable(
+    anomalyTimelineStateService.getContainerWidth$(),
+    anomalyTimelineStateService.getContainerWidth()
+  );
+
+  const columns: Array<EuiBasicTableColumn<AnomalyDetectionAlert>> = [
+    {
+      field: 'ruleName',
+      name: 'Rule name',
```

Comment:
Do these names need to be internationalized?

Reply:
Translated here https://github.com/darnautov/kibana/commit/414ecfa4cb6534f45487f3b6d8c429be574fdc58